protostar/configuration_file/configuration_toml_reader_test.py

```diff
@@ -0,0 +1,184 @@
+
+
+def test_attribute_casing_sensitivity(protostar_toml_path: Path):
+    reader = ConfigurationTOMLReader(protostar_toml_path)
+
+    result = reader.get_attribute(
+        section_name="shared_command_configs",
+        attribute_name="no_color",
+        profile_name="ci",
+        section_namespace="protostar",
+    )
+    result2 = reader.get_attribute(
+        section_name="shared_command_configs",
+        attribute_name="no-color",
+        section_namespace="protostar",
+    )
+    assert result is None
+    assert result2 is None
+
+
+def test_ignoring_attribute_casing(protostar_toml_path: Path):
+    reader = ConfigurationTOMLReader(protostar_toml_path, ignore_attribute_casing=True)
+
+    result = reader.get_attribute(
+        section_name="shared_command_configs",
+        attribute_name="no_color",
+        profile_name="ci",
+        section_namespace="protostar",
+    )
+    result2 = reader.get_attribute(
+        section_name="shared_command_configs",
+        attribute_name="no-color",
+        section_namespace="protostar",
+    )
+    assert result is True
+    assert result2 is False
+
+
+def test_open_file_only_once(protostar_toml_path: Path, mocker: MockerFixture):
+    tomli_mock = mocker.patch(
+        "protostar.configuration_file.configuration_toml_reader.tomli"
+    )
+    tomli_mock.load = mocker.MagicMock()
+    tomli_mock.load.return_value = {}
+
+    reader = ConfigurationTOMLReader(protostar_toml_path)
+
+    reader.get_attribute("_", "_", section_namespace="protostar")
+    reader.get_attribute("__", "__", section_namespace="protostar")
+
+    tomli_mock.load.assert_called_once()
+
+
+def test_exception_on_file_not_found(datadir: Path):
+    with pytest.raises(NoProtostarProjectFoundException):
+        ConfigurationTOMLReader(datadir / "_.toml").get_attribute(
+            "_", "_", section_namespace="protostar"
+        )
+
+
+def test_returning_none_on_attribute_not_found(protostar_toml_path: Path):
+    result = ConfigurationTOMLReader(protostar_toml_path).get_attribute(
+        "shared_command_configs", "undefined_attribute"
+    )
+
+    assert result is None
+
+
+def test_retrieving_section(protostar_toml_path: Path):
+    result = ConfigurationTOMLReader(protostar_toml_path).get_section(
+        "shared_command_configs", section_namespace="protostar"
+    )
+
+    assert result == {"no_color": False}
+
+
+def test_returning_none_on_section_not_found(protostar_toml_path: Path):
+    result = ConfigurationTOMLReader(protostar_toml_path).get_section(
+        "undefined_section"
+    )
+
+    assert result is None
+
+
+def test_extracting_profile_names(protostar_toml_path: Path):
+    result = ConfigurationTOMLReader(protostar_toml_path).get_profile_names()
+
+    assert result == ["ci"]
+
+
+def test_section_starting_with_profile(
+    protostar_toml_path: Path,
+):
+    result = ConfigurationTOMLReader(protostar_toml_path).get_profile_names()
+
+    assert "abc" not in result
```

Comment:
I would be more comfortable to see each unit test work on a case-specific subset of the single fixture that is used here.

Reply:
Test cases are now parametrized.